aiohomekit/zeroconf/__init__.py

```diff
@@ -285,9 +294,7 @@ async def _async_find_data_for_device_id(
 
     try:
         for info in listener.get_data():
-            if not _service_info_is_homekit_device(info):
```

Comment:
I took this out since we are already filtering on HAP_TYPE above and we are looking for a specific id match below